### ISSUE TITLE
Properly handle lambda function bodies in `unnecessary_nesting_linter()`

### ISSUE THREAD

**Issue by salim-b** [r-lib/lintr#2748]
Compare

``` r
code <- "foo <- function(x) {
  x + 1
}"
writeLines(code)
#> foo <- function(x) {
#>   x + 1
#> }
lintr::lint(
    text = code,
    linters = lintr::unnecessary_nesting_linter()
)
#> ℹ No lints found.
```

<sup>Created on 2025-02-13 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

vs.

``` r
code <- "foo <- \\(x) {
  x + 1
}"
writeLines(code)
#> foo <- \(x) {
#>   x + 1
#> }
lintr::lint(
    text = code,
    linters = lintr::unnecessary_nesting_linter()
)
#> <text>:1:13: warning: [unnecessary_nesting_linter] Reduce the nesting of this statement by removing the braces {}.
#> foo <- \(x) {
#>             ^
```

<sup>Created on 2025-02-13 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Linting this case should be the responsibility of https://github.com/r-lib/lintr/pull/2240/, instead.
